src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -536,6 +537,42 @@ static std::optional<std::vector<ASTPtr>> getExpressionsOfUpdatedNestedSubcolumn
     return res;
 }
 
```

Comment:
This code copies logic from PlannerJoinTree.cpp. You need to put it in common place

Reply:
I've moved the common logic to the AddingTableNameVirtualColumnStep.cpp file.